extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php

```diff
@@ -74,12 +77,16 @@ public function execute( $subpage ) {
 
 		#generate what the token SHOULD be
 		$shouldToken = wfGenerateUnsubToken($email, $timestamp);
-		if( $token != $shouldToken ) {
+		if( !hash_equals( $shouldToken, $token ) ) {
 			$wgOut->addWikiMsg( 'unsubscribe-badtoken' );
 			// $wgOut->addHTML("shouldtoken={$shouldToken}\n"); #DEVL (remove before release)
 			return;
 		}
 
+		// PLATFORM-2389: request token is checked above
+		// tell CSRFDetector that we're fine here
+		CSRFDetector::onUserMatchEditToken();
+
 		#does the non-blank email they gave us look like an email?
 		if( Sanitizer::validateEmail( $email ) == false ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1125)